src/sentry/api/endpoints/team_members.py

```diff
@@ -1,25 +1,54 @@
-from django.db.models import Q
+from django.db.models import Q, prefetch_related_objects
 from rest_framework.request import Request
 from rest_framework.response import Response
 
 from sentry.api.bases.team import TeamEndpoint
-from sentry.api.serializers import serialize
-from sentry.models import InviteStatus, OrganizationMember
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models import InviteStatus
+from sentry.models.organizationmemberteam import OrganizationMemberTeam
+
+
+@register(OrganizationMemberTeam)
+class DetailedOrganizationMemberTeamSerializer(Serializer):
+    def __init__(self, *args, **kwargs):
+        self.team = kwargs.pop("team", None)
+        super().__init__(*args, **kwargs)
+
+    def get_attrs(self, item_list, user, **kwargs):
+        prefetch_related_objects(item_list, "organizationmember")
+        prefetch_related_objects(item_list, "organizationmember__user")
+        attrs = {}
+        for org_member_team in item_list:
+            attrs[org_member_team] = {"org_member": org_member_team.organizationmember}
+        return attrs
+
+    def serialize(self, obj, attrs, user):
+        org_member = serialize(attrs["org_member"])
```

Comment:
Serializing this here can cause N+1. Take a look at `OrganizationMemberSerializer` to understand why. It does a bunch of IO in `get_attrs`, so if you serialize each of these individually it'll make a query per row.

Ideally, what you should do here is in `get_attrs`, get a set of all the unique `organization_member` rows out of the `org_member_team` items, and call `serialize` on all of them there in one call. Then include the serialized results in the `attrs` dictionary instead.